lib/sycamore/sycamore/query/execution/sycamore_executor.py

```diff
@@ -91,10 +91,11 @@ def process_node(self, logical_node: LogicalOperator, query_id: str) -> Any:
         # Process dependencies
         if logical_node.dependencies:
             for dependency in logical_node.dependencies:
+                assert isinstance(dependency, LogicalOperator)
```

Comment:
want to just assert this in LogicalOperator instead

Reply:
You mean by overriding the `dependencies` property? Currently this is provided by `Node`, not `LogicalOperator`, so the only place we can assert is here.

Reply:
I thought you can do

```
class LogicalOperator..:

def __init__:
  super()
  # assert dependencies are LogicalOperators
```

anyway not a blocker

Reply:
The problem is that one can reassign the values of any of the dependencies after the constructor is called, since the object as a whole is not immutable.

Reply:
Ahh gotcha